tests/test_routing.py

```diff
@@ -1057,3 +1058,28 @@ def test_error_message_suggestion():
     with pytest.raises(r.BuildError) as excinfo:
         adapter.build("world", {"id": 2}, method="POST")
     assert "Did you mean to use methods ['GET', 'HEAD']?" in str(excinfo.value)
+
+
+def test_no_memory_leak_from_Rule_builder():
+    """See #1520"""
+
+    # generate a bunch of objects that *should* get collected
+    for _ in range(100):
+        r.Map([r.Rule("/a/<string:b>")])
+
+    # ensure that the garbage collection has had a chance to collect cyclic
+    # objects
+    for _ in range(5):
```

Comment:
`gc.collect()` returns the number of objects collected so i _think_ this can be replaced by

```
while gc.collect():
    pass
````

Reply:
That would cause the test to never stop instead of failing though. 5 seems like a safe bet.

Reply:
> 
> 
> `gc.collect()` returns the number of objects collected so i _think_ this can be replaced by
> 
> ```
> while gc.collect():
>     pass
> ```

Not necessarily, the first generation could collect zero objects (fast gc) and then later generations could collect cyclic gc